src/mon/FSCommands.cc

```diff
@@ -260,23 +259,18 @@ class SetHandler : public FileSystemCommandHandler
       }
 
       if (n <= 0) {
-        ss << "You must specify at least one MDS";
+        ss << "max_mds cannot be negative or zero";
         return -EINVAL;
-      }
-
-      if (!fs->mds_map.allows_multimds() && n > fs->mds_map.get_max_mds() &&
-	  n > 1) {
-	ss << "multi-MDS clusters are not enabled; set 'allow_multimds' to enable";
-	return -EINVAL;
-      }
-      if (n > MAX_MDS) {
+      } else if (n > MAX_MDS) {
         ss << "may not have more than " << MAX_MDS << " MDS ranks";
```

Comment:
could we also point the user to the right way to increase this value?

Reply:
They cannot currently. It is hard-coded limit used to bound the range of metadata object names in the metadata pool.